src/main/java/io/github/tetherless_world/jena_elasticsearch/ElasticsearchGraph.java

```diff
@@ -56,11 +62,44 @@ public void performAdd(Triple t) {
         jsonMap.put("object", getNodeContent(t.getObject()));
 
         try {
-            IndexRequest request = new IndexRequest(this.name).source(jsonMap);
+            final IndexRequest request = new IndexRequest(this.name).source(jsonMap);
             this.client.index(request, RequestOptions.DEFAULT);
         } catch (IOException e) {
             logger.error("Error indexing triple: {}", t, e);
-            e.printStackTrace();
+        }
+
+        blockUntilAdded(t);
+    }
+
+    private void blockUntilAdded(Triple t) {
+        final RefreshRequest refreshRequest = new RefreshRequest(this.name);
+        final SearchRequest searchRequest = new SearchRequest();
+        final SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+
+        searchSourceBuilder.query(this.constructExactTripleMatchingQuery(t));
+        searchRequest.indices(this.name);
+        searchRequest.source(searchSourceBuilder);
+
+        // Try 10 times to refresh Elasticsearch
+        for (int i=0; i<10; ++i) {
```

Comment:
You can factor this loop out into a separate method that takes a java.util.function.Predicate to do the test.